cargo-dist/src/tasks.rs

```diff
@@ -1002,16 +1004,13 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
         let mut package_configs = vec![];
 
         for (pkg_idx, package) in workspaces.all_packages() {
-            let mut package_metadata = config::parse_metadata_table_or_manifest(
-                &package.manifest_path,
-                package.dist_manifest_path.as_deref(),
-                package.cargo_metadata_table.as_ref(),
-            )?;
+            let mut package_metadata =
+                config::try_load_package_config(package.dist_manifest_path.as_ref())?;
             package_configs.push(app_config(
                 workspaces,
                 pkg_idx,
                 workspace_layer.clone(),
-                package_metadata.to_toml_layer(false),
+                package_metadata.clone(), // fixme: this seems suspect since we call seemingly-mutating functions on it later
```

Comment:
this is _probably_ fine because `package_metadata.to_toml_layer(false)` returns a new object anyway?